frontend/src/features/side_window/beacon_malfunctions/BeaconMalfunctionCard.tsx

```diff
@@ -0,0 +1,253 @@
+  isShowed,
+  updateVesselStatus,
+  verticalScrollRef
+}) {
+  const dispatch = useAppDispatch()
+  const vesselStatus = vesselStatuses.find(_vesselStatus => _vesselStatus.value === beaconMalfunction?.vesselStatus)
+  const bodyRef = useRef<HTMLDivElement>(null)
+  const wrapperRef = useRef<HTMLDivElement>(null)
+
+  const endOfBeaconMalfunctionReason = useMemo(
+    () => endOfBeaconMalfunctionReasons[beaconMalfunction?.endOfBeaconMalfunctionReason],
+    [beaconMalfunction]
+  )
+
+  useEffect(() => {
+    if (bodyRef.current && vesselStatus && vesselStatus.color && beaconMalfunction?.id) {
+      // TODO Use styled-component and avoid useEffect to update these elements style.
+      const selectElement = bodyRef.current.querySelector('.rs-picker-select') as HTMLElement
+      if (selectElement?.style) {
+        selectElement.style.background = vesselStatus.color
+      }
+
+      const toggleElement = bodyRef.current.querySelector(
+        '*[data-cy="side-window-beacon-malfunctions-vessel-status"]'
+      ) as HTMLElement
+      if (toggleElement?.style) {
+        toggleElement.style.color = vesselStatus.textColor
+      }
+    }
+  }, [vesselStatus, beaconMalfunction])
+
+  useEffect(() => {
+    if (isShowed && beaconMalfunction && wrapperRef.current) {
+      wrapperRef.current.scrollIntoView({
+        behavior: 'smooth',
+        block: 'nearest'
+      })
+    }
+  }, [isShowed, beaconMalfunction])
+
+  return (
+    <Wrapper
+      ref={wrapperRef}
+      data-cy="side-window-beacon-malfunctions-card"
+      style={wrapperStyle(
+        verticalScrollRef?.current?.scrollHeight > verticalScrollRef?.current?.clientHeight,
+        isDragging,
+        isDroppedId,
+        beaconMalfunction?.id,
+        activeBeaconId,
+        endOfBeaconMalfunctionReason
+      )}
+    >
+      <Header style={headerStyle}>
+        <Row style={rowStyle(true, 8)}>
+          <Id data-cy="side-window-vessel-id" style={idStyle}>
+            #{beaconMalfunction?.id} - {getBeaconCreationOrModificationDate(beaconMalfunction)}
+          </Id>
+          <ShowIcon
+            alt="Voir sur la carte"
+            // TODO Fix the TS error when an action returns a Promise
+            // @ts-ignore
+            onClick={() => dispatch(showVesselFromBeaconMalfunctionsKanban(beaconMalfunction, false))}
+            src={`${baseUrl}/View_on_map.png`}
+            style={showIconStyle}
+          />
+        </Row>
+        <Row style={rowStyle(false, 4)}>
+          {beaconMalfunction?.flagState ? (
+            <Flag src={`${baseUrl}/flags/${beaconMalfunction?.flagState.toLowerCase()}.svg`} style={flagStyle} />
+          ) : null}
+          <VesselName
+            className="hover-border"
+            data-cy="side-window-beacon-malfunctions-card-vessel-name"
+            // TODO Fix the TS error when an action returns a Promise
+            // @ts-ignore
+            onClick={() => dispatch(openBeaconMalfunctionInKanban({ beaconMalfunction }))}
```

Comment:
This is why it's not so good to directly put the logic inside the props. Here the issue can simply be either add brackets or better: put the code in a callback above.